topology/graph/client.go

```diff
@@ -168,9 +168,11 @@ func (c *AsyncClient) AddListener(l EventListener) {
 }
 
 func (c *AsyncClient) Disconnect() {
+	if c.IsConnected() {
+		c.quit <- true
+		c.wg.Wait()
+	}
 	c.running.Store(false)
```

Comment:
c.running.Store(false) need to be set first
the real fix is to call close(c.quit) to broadcast to all thread

c.running.Store(false)
close(c.quit)
c.wg.Wait()